cfn-resources/teams/cmd/resource/resource.go

```diff
@@ -381,28 +381,28 @@ func Delete(req handler.Request, prevModel *Model, currentModel *Model) (handler
 				HandlerErrorCode: cloudformation.HandlerErrorCodeInternalFailure,
 			}, nil
 		}
-	} else {
-		// remove from organization
-		err := removeFromOrganization(atlasV2, currentModel)
-		if err != nil {
-			// if team is assigned to project then first delete from project
-			if atlasv2.IsErrorCode(err, "CANNOT_DELETE_TEAM_ASSIGNED_TO_PROJECT") {
-				if err := removeFromProject(atlasV2, currentModel); err != nil {
-					return handler.ProgressEvent{
-						OperationStatus:  handler.Failed,
-						Message:          "Unable to Delete",
-						HandlerErrorCode: cloudformation.HandlerErrorCodeInternalFailure,
-					}, nil
-				}
+	}
 
-				// remove from organization if successfully deleted from project
-				if err := removeFromOrganization(atlasV2, currentModel); err != nil {
-					return handler.ProgressEvent{
-						OperationStatus:  handler.Failed,
-						Message:          "Unable to Delete",
-						HandlerErrorCode: cloudformation.HandlerErrorCodeInternalFailure,
-					}, nil
-				}
+	// remove from organization
+	err := removeFromOrganization(atlasV2, currentModel)
+	if err != nil {
+		// if team is assigned to project then first delete from project
+		if atlasv2.IsErrorCode(err, "CANNOT_DELETE_TEAM_ASSIGNED_TO_PROJECT") {
+			if err := removeFromProject(atlasV2, currentModel); err != nil {
+				return handler.ProgressEvent{
+					OperationStatus:  handler.Failed,
+					Message:          "Unable to Delete",
+					HandlerErrorCode: cloudformation.HandlerErrorCodeInternalFailure,
+				}, nil
+			}
+
+			// remove from organization if successfully deleted from project
+			if err := removeFromOrganization(atlasV2, currentModel); err != nil {
+				return handler.ProgressEvent{
```

Comment:
does an error here will leave in an inconsitent/unexpected state as it's remove from the project but not from the org?    i suppose it's too much to "compensate" and add it to the project again to keep this operation atomic.

Reply:
nice catch @lantoli 
instead we could retry and communicate the failure in the message properly due to org association so users can look into it?

Reply:
I have adjusted the error message in this case so that user is aware of this side effect even though operation is not successful.